src/main/java/com/exasol/udfstartuptimeimprover/UdfStartUpTimeImprover.java

```diff
@@ -0,0 +1,171 @@
+                    .mitigation("Make sure that you set '" + PARAMETER_CONNECTION
+                            + "' to the name of a connection definition for BucketFS.")
+                    .toString(), exception);
+        }
+    }
+
+    private static String readStringParameter(final ExaIterator iterator, final String parameterName) {
+        try {
+            return Objects.requireNonNull(iterator.getString(parameterName));
+        } catch (final ExaIterationException | ExaDataTypeException | NullPointerException exception) {
+            throw new IllegalArgumentException(ExaError.messageBuilder("E-USTI-12")
+                    .message("Could not read required parameter {{name}}.", parameterName)
+                    .mitigation("Make sure that you provide that parameter and that it's of type VARCHAR type.")
+                    .toString(), exception);
+        }
+    }
+
+    private static int readIntParameter(final ExaIterator iterator, final String parameterName) {
+        try {
+            return Objects.requireNonNull(iterator.getInteger(parameterName));
+        } catch (final ExaIterationException | ExaDataTypeException | NullPointerException exception) {
+            throw new IllegalArgumentException(ExaError.messageBuilder("E-USTI-15")
+                    .message("Could not read required parameter {{name}}.", parameterName)
+                    .mitigation("Make sure that you provide that parameter and that it's of type INTEGER type.")
+                    .toString(), exception);
+        }
+    }
+
+    private static void runClassDumpGeneration(final List<String> jars, final Path pathToClassList) {
+        try {
+            final List<String> commandParts = new ArrayList<>(
+                    List.of("java", "-Xshare:dump", "-XX:SharedClassListFile=" + pathToClassList,
+                            "-XX:SharedArchiveFile=" + DUMP_IN_TMP, "--class-path"));
+            commandParts.addAll(jars);
+            final Process process = Runtime.getRuntime().exec(commandParts.toArray(String[]::new));
+            final int exitCode = process.waitFor();
+            assertExitCode(process, exitCode);
+        } catch (final InterruptedException exception) {
+            Thread.currentThread().interrupt();
+            throw new IllegalStateException(ExaError.messageBuilder("E-USTI-11")
+                    .message("Interrupted while building class archive.").toString(), exception);
+        } catch (final IOException exception) {
+            throw new IllegalStateException(
+                    ExaError.messageBuilder("E-USTI-10").message("Building class archive failed").toString(),
+                    exception);
+        }
+    }
+
+    private static void assertExitCode(final Process process, final int exitCode) throws IOException {
+        if (exitCode != 0) {
+            final byte[] output = process.getInputStream().readAllBytes();
```

Comment:
We had an issue in pk with starting a process that writes a lot of data to stdout. But if this doesn't write too much it should be fine.

Reply:
I had the same thought.